src/app/components/core/configuration-manager/list-type-configuration/list-type-configuration.component.ts

```diff
@@ -69,8 +102,48 @@ export class ListTypeConfigurationComponent implements OnInit {
           return num;
         });
       }
-      this.form.get(this.configuration.key)?.patchValue(JSON.stringify(value))
-      this.formState.emit(this.listItems.valid);
+      if(this.configuration.items == 'object') {
+        let listValues = this.generateListValuesArray(value);
+        this.changedConfig.emit({ [this.configuration.key]: JSON.stringify(listValues) });
+      }
+      else{
+        this.changedConfig.emit({ [this.configuration.key]: JSON.stringify(value) });
+      }
+      this.formStatusEvent.emit({'status': this.listItems.valid, 'group': this.group});
+      // this.form.get(this.configuration.key)?.patchValue(JSON.stringify(value))
+      // this.formState.emit(this.listItems.valid);
     })
   }
+
+  getChangedConfiguration(index: string, propertyChangedValues: any) {
+    for (let [ind, val] of this.listItems.value.entries()) {
+      let obj = JSON.parse(val);
+      for (let property in obj) {
+        if(ind==index && property == Object.keys(propertyChangedValues)[0]) {
+          obj[property].value = Object.values(propertyChangedValues)[0];
+        }
```

Comment:
if(ind==index && property == Object.keys(propertyChangedValues)[0])  this check need more clarification. why index 0 key compare only?

Reply:
Here index is required to change corresponding list item and index 0 key is compared because `propertyChangedValues` object will always have one key value pair because this method will be called for each change in form.